priority/management/commands/calculate-priority.py

```diff
@@ -0,0 +1,121 @@
+        SCW = pd.merge(SCW,gpu_sum, on='Account', how='outer').fillna(0)
+        #give all new entries an Ap of 50000
+        #SCW.update(default_Ap,overwrite=False,filter_func=new_data)
+        SCW.Ap.loc[SCW.Ap==0]=50000
+        SCW['Cpu_New']=SCW['cpu_sum'].subtract(SCW['CPU_hours'])
+        SCW['Gpu_New']=SCW['gpu_sum'].subtract(SCW['GPU_hours'])
+        
+        
+        #calculate parameter such that K*log(Ap) has a max defined by QOS_levels
+        Max_Ap=SCW['Ap'].max()
+        k=QOS_levels/np.log10(Max_Ap)
+        #check if previous run was prioritezed and if so add the time since last sactdump to total priortized time
+        P=SCW['Qos']>0
+        Prioritsed=SCW[P]
+        Prioritsed['P_CPU'].add(Prioritsed['Cpu_New'])
+        Prioritsed['P_GPU'].add(Prioritsed['Gpu_New'])
+        
+        SCW.update(Prioritsed)
+        SCW['CPU_hours']=SCW['CPU_hours']+SCW['Cpu_New']
+        SCW['GPU_hours']=SCW['GPU_hours']+SCW['Gpu_New']
+        #calculate Priority
+        Priority =(SCW['Ap']-(SCW['P_CPU'] + SCW['P_GPU']*40)).to_frame(name='Priority_Sum').astype(int)
+        Priority['QOS']=SCW['Ap'].map(lambda a : k*np.log10(a)).astype(int)
+        Priority['QOS']= Priority['QOS'].where(Priority['Priority_Sum']>=0,0)
+        Priority['Account']=SCW['Account']
+        
+        
+        #create dataframe for output to django
+        COGS_DB = pd.DataFrame(Priority['Account'], columns=['Account'])
+        COGS_DB['QOS']=Priority['QOS']
+        COGS_DB['CPU_hours']=SCW['CPU_hours']
+        COGS_DB['GPU_hours']=SCW['GPU_hours']
+        COGS_DB['P_CPU']=SCW['P_CPU']
+        COGS_DB['P_GPU']=SCW['P_GPU']
+        COGS_DB['Ap']=SCW['Ap']
+        COGS_DB['Date']=pd.date_range(start=Today, end=Today,periods=len(SCW.index))
+        #output Acoount and QOS to pipe seperated csv file 
+        Priority.to_csv(path_or_buf='~/Documents/Cogs3/cogs3/priority/fixtures/QOS_output.csv',sep='|',columns=['Account','QOS'],index=False)
+  #chack put in place to prevent writing data twice on the same day
+        assert (not slurm_priority.objects.filter(Date__year=Today.year, Date__month=Today.month, Date__day=Today.day).exists()),'Data for today already exists in django database'
+        # Write data to django database
+        COGS_DB.to_sql('priority_slurm_priority',index=False,con=engine,if_exists='append')
+        engine.execute("SELECT * FROM priority_slurm_priority").fetchall()
+        
+def caculate_AP():
+    from django.core.management.base import BaseCommand, CommandError
+    from django.conf import settings
+    from django.db import models
+    from project.models import Project, ProjectSystemAllocation
+    from funding.models import Attribution
+    from priority.models import slurm_priority
+    from datetime import date, timedelta
+#list all projects that are on sunbird
+    yesterday=date.today()-timedelta(1)
+    on_sunbird=[]
+    Sunbird=ProjectSystemAllocation.objects.exclude(system=1)
```

Comment:
I'd like to avoid hardcoding "Sunbird" into the application where possible. Elsewhere we've done this by adding feature flags at an Institution level for features institutions want to opt into. In this case, we can apply the same AP calculation to everyone, since AP for non-Sunbird users will never be surfaced anywhere. (This also means that Cardiff and Bangor users temporarily running on Sunbird for whatever reason will be incorporated into the priority system, even if not being able to add attributions means that they will only get the initial 50k APs.)